### PR TITLE
[GDR-1975] data.frame-related functions

### PR DESCRIPTION
**Pull request by j-smola**
# Description
## What changed?
Related JIRA issue: GDR-1975

## Why was it changed?
Change for for data format consistency (data.table)

# Checklist for sustainable code base
- [ ] I added tests for any code changed/added
- [ ] I added documentation for any code changed/added
- [ ] I made sure naming of any new functions is self-explanatory and consistent

# Logistic checklist
- [ ] Package version bumped
- [ ] Changelog updated

# Screenshots (optional)

### CODE REVIEW

File: tests/testthat/test-validate_identifiers.R [hunk 1]
```diff
@@ -59,8 +59,8 @@ test_that(".check_polymapped_identifiers works as expected", {
   id_map <- as.list(LETTERS[seq(nids)])
   names(id_map) <- exp_one_ids
 
-  df <- as.data.frame(matrix(rep(0, nids * nrow), nrow = nrow, ncol = nids))
-  names(df) <- id_map
+  df <- data.table::data.table(matrix(rep(0, nids * nrow), nrow = nrow, ncol = nids))
```
Comment (bczech) [inline tests/testthat/test-validate_identifiers.R:62]:
```suggestion
  df <- data.table::as.data.table(matrix(rep(0, nids * nrow), nrow = nrow, ncol = nids))
```

**Review by gladkia — APPROVED** [review]

File: tests/testthat/test-convert_mae_se_assay_to_dt.R [hunk 2]
```diff
@@ -54,7 +54,7 @@ test_that("merge_metrics argument of assay_to_dt works as expected", {
   headers <- get_header("response_metrics")
   m <- 20
   n <- length(headers)
-  metrics <- as.data.frame(matrix(runif(m * 2 * n), nrow = m * 2, ncol = n))
+  metrics <- data.table::data.table(matrix(runif(m * 2 * n), nrow = m * 2, ncol = n))
   colnames(metrics) <- headers
```
Comment (bczech) [inline tests/testthat/test-convert_mae_se_assay_to_dt.R:58]:
```suggestion
    data.table::setnames(metrics, headers```

File: tests/testthat/test-convert_mae_se_assay_to_dt.R [hunk 3]
```diff
@@ -54,7 +54,7 @@ test_that("merge_metrics argument of assay_to_dt works as expected", {
...
   metrics$dr_metric <- rep(c("RV", "GR"), m)
```
Comment (bczech) [inline tests/testthat/test-convert_mae_se_assay_to_dt.R:59]:
```suggestion
  metrics[, dr_metric := rep(c("RV", "GR"), m)]
```

File: tests/testthat/test-json_convert.R [hunk 4]
```diff
@@ -14,7 +14,7 @@ test_that(".standardize_column_names works as expected", {
 test_that(".convert_element_metadata_to_json works as expected", {
   m <- 100
   n <- 26
-  data <- data.table::setDT(as.data.frame(matrix(rep(seq(m), n), nrow = m)))
+  data <- data.table::data.table(matrix(rep(seq(m), n), nrow = m))
   shuffle <- sample(1:n, n)
   names(data) <- LETTERS[shuffle]
```
Comment (bczech) [inline tests/testthat/test-json_convert.R:19]:
```suggestion
  data.table::setnames(data, LETTERS[shuffle])
```

File: tests/testthat/test-validate_identifiers.R [hunk 5]
```diff
@@ -59,8 +59,8 @@ test_that(".check_polymapped_identifiers works as expected", {
...
+  colnames(df) <- unlist(id_map)
```
Comment (bczech) [inline tests/testthat/test-validate_identifiers.R:63]:
```suggestion
  data.table::setnames(df, unlist(id_map))
```

File: tests/testthat/test-validate_identifiers.R [hunk 6]
```diff
@@ -90,8 +90,8 @@ test_that("gDRutils:::.modify_polymapped_identifiers works as expected", {
   id_map <- as.list(LETTERS[seq(nids)])
   names(id_map) <- exp_one_ids
 
-  df <- as.data.frame(matrix(rep(0, nids * nrow), nrow = nrow, ncol = nids))
-  names(df) <- id_map
+  df <- data.table::data.table(matrix(rep(0, nids * nrow), nrow = nrow, ncol = nids))
```
Comment (bczech) [inline tests/testthat/test-validate_identifiers.R:93]:
```suggestion
  df <- data.table::as.data.table(matrix(rep(0, nids * nrow), nrow = nrow, ncol = nids))
```

File: tests/testthat/test-validate_identifiers.R [hunk 7]
```diff
@@ -90,8 +90,8 @@ test_that("gDRutils:::.modify_polymapped_identifiers works as expected", {
...
+  colnames(df) <- unlist(id_map)
```
Comment (bczech) [inline tests/testthat/test-validate_identifiers.R:94]:
```suggestion
  data.table::setnames(df, unlist(id_map))
```

File: tests/testthat/test-validate_identifiers.R [hunk 8]
```diff
@@ -126,8 +126,8 @@ test_that("validate_identifiers works as expected", {
   ids <- as.list(LETTERS[seq(nids)])
   names(ids) <- exp_one_ids
 
-  df <- as.data.frame(matrix(rep(0, nids * nrow), nrow = nrow, ncol = nids))
-  names(df) <- ids
+  df <- data.table::data.table(matrix(rep(0, nids * nrow), nrow = nrow, ncol = nids))
```
Comment (bczech) [inline tests/testthat/test-validate_identifiers.R:129]:
```suggestion
  df <- data.table::as.data.table(matrix(rep(0, nids * nrow), nrow = nrow, ncol = nids))
```

File: tests/testthat/test-validate_identifiers.R [hunk 9]
```diff
@@ -126,8 +126,8 @@ test_that("validate_identifiers works as expected", {
...
+  colnames(df) <- unlist(ids)
```
Comment (bczech) [inline tests/testthat/test-validate_identifiers.R:130]:
```suggestion
  data.table::setnames(df, unlist(ids))
```

File: tests/testthat/test-convert_mae_se_assay_to_dt.R [hunk 10]
```diff
@@ -54,7 +54,7 @@ test_that("merge_metrics argument of assay_to_dt works as expected", {
   headers <- get_header("response_metrics")
   m <- 20
   n <- length(headers)
-  metrics <- as.data.frame(matrix(runif(m * 2 * n), nrow = m * 2, ncol = n))
+  metrics <- data.table::data.table(matrix(runif(m * 2 * n), nrow = m * 2, ncol = n))
```
[the remark this answers is not in the document]
Comment (j-smola) [reply tests/testthat/test-convert_mae_se_assay_to_dt.R:57]: ✅

File: tests/testthat/test-json_convert.R [hunk 11]
```diff
@@ -14,7 +14,7 @@ test_that(".standardize_column_names works as expected", {
 test_that(".convert_element_metadata_to_json works as expected", {
   m <- 100
   n <- 26
-  data <- data.table::setDT(as.data.frame(matrix(rep(seq(m), n), nrow = m)))
+  data <- data.table::data.table(matrix(rep(seq(m), n), nrow = m))
```
[the remark this answers is not in the document]
Comment (j-smola) [reply tests/testthat/test-json_convert.R:17]: ✅

File: tests/testthat/test-validate_identifiers.R [hunk 12]
```diff
@@ -10,8 +10,8 @@ test_that(".check_required_identifiers works as expected", {
   ids <- as.list(LETTERS[seq(nids)])
   names(ids) <- req_ids
 
-  df <- as.data.frame(matrix(rep(0, nids * nrow), nrow = nrow, ncol = nids))
-  names(df) <- ids
+  df <- data.table::data.table(matrix(rep(0, nids * nrow), nrow = nrow, ncol = nids))
```
[the remark this answers is not in the document]
Comment (j-smola) [reply tests/testthat/test-validate_identifiers.R:13]: ✅

File: tests/testthat/test-validate_identifiers.R (hunk 1 above)
Comment (j-smola) [reply tests/testthat/test-validate_identifiers.R:62]: ✅

File: tests/testthat/test-validate_identifiers.R [hunk 13]
```diff
@@ -10,8 +10,8 @@ test_that(".check_required_identifiers works as expected", {
...
+  colnames(df) <- unlist(ids)
```
[the remark this answers is not in the document]
Comment (j-smola) [reply tests/testthat/test-validate_identifiers.R:14]: ✅

File: tests/testthat/test-validate_identifiers.R (hunk 5 above)
Comment (j-smola) [reply tests/testthat/test-validate_identifiers.R:63]: ✅

File: tests/testthat/test-validate_identifiers.R (hunk 9 above)
Comment (j-smola) [reply tests/testthat/test-validate_identifiers.R:130]: ✅

File: tests/testthat/test-validate_identifiers.R (hunk 8 above)
Comment (j-smola) [reply tests/testthat/test-validate_identifiers.R:129]: ✅

File: tests/testthat/test-validate_identifiers.R (hunk 7 above)
Comment (j-smola) [reply tests/testthat/test-validate_identifiers.R:94]: ✅

File: tests/testthat/test-validate_identifiers.R (hunk 6 above)
Comment (j-smola) [reply tests/testthat/test-validate_identifiers.R:93]: ✅

File: tests/testthat/test-json_convert.R (hunk 4 above)
Comment (j-smola) [reply tests/testthat/test-json_convert.R:19]: ✅

File: tests/testthat/test-convert_mae_se_assay_to_dt.R (hunk 3 above)
Comment (j-smola) [reply tests/testthat/test-convert_mae_se_assay_to_dt.R:59]: ✅

File: tests/testthat/test-convert_mae_se_assay_to_dt.R (hunk 2 above)
Comment (j-smola) [reply tests/testthat/test-convert_mae_se_assay_to_dt.R:58]: ✅

File: R/df_to_bm_assay.R [hunk 14]
```diff
@@ -49,57 +43,10 @@ df_to_bm_assay <-
...
-.get_treated_conditions <-
-  function(drug_data) {
-    # Assertions:
-    stopifnot(any(inherits(drug_data, "data.table")))
-    .untreated_tag_patterns <- vapply(get_env_identifiers("untreated_tag"), sprintf, fmt = "^%s$", character(1))
-    .untreatedDrugNameRegex <- paste(.untreated_tag_patterns, collapse = "|")
-    drugnames <- tolower(data.table::setDT(drug_data)[, get_env_identifiers("drugname")])
-    drug_data[!grepl(.untreatedDrugNameRegex, drugnames), "name_"]
+    
+    return(bm)
```
Comment (bczech) [inline R/df_to_bm_assay.R:51]: Do we need to use `return` now?

File: R/fit_curves.R [hunk 15]
```diff
@@ -258,7 +257,8 @@ logisticFit <-
                       priors = priors, lower = lower, force_fit = force_fit, x_0 = x_0, cap = cap, 
                       concs = concs, controls = controls, range_conc = range_conc, pcutoff = pcutoff, 
                       capping_fold = capping_fold, mean_norm_value = mean_norm_value)
-    data.table::data.table(data.frame(out))
+    
+    data.table::setDT(out)
```
Comment (bczech) [inline R/fit_curves.R:261]: add new line after it with `out`

File: R/fit_curves.R [hunk 16]
```diff
@@ -512,11 +512,14 @@ has_dups <- function(vec) {
 #' @importFrom stats aggregate
 average_dups <- function(df, col) {
   stopifnot(col %in% colnames(df))
-  aggregate(df[colnames(df) != col],
-    by = list(concs = df[[col]]),
-    FUN = function(x) {
-      mean(x, na.rm = TRUE)
-    }
+  agg_col <- colnames(df)[colnames(df) != col]
+  data.table::setDT(
```
Comment (bczech) [inline R/fit_curves.R:516]: `as.data.table`

File: R/df_to_bm_assay.R (hunk 14 above)
Comment (j-smola) [reply R/df_to_bm_assay.R:51]: ✅

File: R/fit_curves.R (hunk 15 above)
Comment (j-smola) [reply R/fit_curves.R:261]: added ✅

File: R/fit_curves.R (hunk 16 above)
Comment (j-smola) [reply R/fit_curves.R:516]: ✅

File: R/fit_curves.R [hunk 17]
```diff
@@ -259,6 +259,7 @@ logisticFit <-
                       capping_fold = capping_fold, mean_norm_value = mean_norm_value)
     
     data.table::setDT(out)
+
```
Comment (bczech) [inline R/fit_curves.R:262]: here should be `out` not an empty line, right?

**Review by bczech — APPROVED** [review]

**Pull request merged**